addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -76,6 +80,8 @@
 
     private static final int HEARTBEAT_INTERVAL = 30;
 
+    private ScheduledExecutorService buttonExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Yes they can be long running calls and potentially block,  for example If I send "1 2 3 enter" quickly, I want each to execute without sucking up all threads or blocking the main thread, but I also only need one to execute after another and not in parallel